### PR TITLE
Update the LHESource from fix in 75.

### PR DESCRIPTION
**Pull request by parbol**
This pull request contains the update that was done in 75 to solve the issue where only 3 files were being read by LHESource.

### CODE REVIEW

**Comment by bendavid** [comment]
@parbol the change doesn't look quite the same as the one in https://github.com/cms-sw/cmssw/pull/7795.
In particular there seems to be an extra 
"partonLevel = reader->next(&newFileOpened);"

Did I get it wrong?  What is the reason for the difference?

**Comment by parbol** [comment]
Hi Josh,

The reason for this catch is that just before pushing I changed branch and that line got merged without me noticing. It's obviously a mistake. I have deleted it and updated the code, it should be fine now.

File: GeneratorInterface/LHEInterface/plugins/LHESource.cc [hunk 1]
```diff
@@ -64,8 +64,19 @@ void LHESource::nextEvent()
 		return;
         }
 
-	bool newFileOpened = false;
-	partonLevel = reader->next(&newFileOpened);
+	//bool newFileOpened = false;
```
Comment (bendavid) [inline GeneratorInterface/LHEInterface/plugins/LHESource.cc]: please just remove these two lines rather than commenting them.

**Comment by bendavid** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by ferencek** [comment]
The PR description suggests there was a problem with reading more than 3 LHE files in a single LHESource but from my tests in CMSSW_7_1_16_patch1 the following configuration (LHE files contain 50k events each)
```python
process.maxEvents = cms.untracked.PSet(
    input = cms.untracked.int32(100)
)

# Input source
process.source = cms.Source("LHESource",
    fileNames = cms.untracked.vstring(
        'root://cmsxrootd.fnal.gov//store/user/ferencek/LHE/Stop2ToStop1H_13TeV-madgraph/stop2_1000.0_0_decayed_1000006_200__1000022_25.lhe',
        'root://cmsxrootd.fnal.gov//store/user/ferencek/LHE/Stop2ToStop1H_13TeV-madgraph/stop2_1000.0_0_decayed_1000006_400__1000022_225.lhe'
    ),
    skipEvents = cms.untracked.uint32(50100)
)
```
would result in a fatal exception
```
----- Begin Fatal Exception 05-Jun-2015 12:17:12 CDT-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1
   [1] Calling beginRun for unscheduled module Pythia8HadronizerFilter/'generator'
Exception Message:
Principal::getByLabel: Found zero products matching all criteria
Looking for type: LHERunInfoProduct
Looking for module label: source
Looking for productInstanceName: 
Looking for process: LHEFile
----- End Fatal Exception -------------------------------------------------
```
In CMSSW_7_1_16_patch2, after this PR got included, it all runs as expected. So it seems there was already a problem with reading more than 1 LHE file. Can an expert please comment?

Also note that the above exception causes CRAB jobs to fail in case they attempt to read events beyond the first file. Interestingly enough, if a job started reading events at the end of the first file and continued to the second, that worked just fine.

**Pull request merged**